### DISCUSSION TITLE
Plans for add KHR_animation_pointer?

### DISCUSSION THREAD

**Discussion by arpu** [zeux/meshoptimizer#694] [Q&A]
This is now merged in git Khronos
https://github.com/KhronosGroup/glTF/tree/main/extensions/2.0/Khronos/KHR_animation_pointer

Blender export 
https://github.com/KhronosGroup/glTF-Blender-IO/pull/2061

**Comment by zeux** [comment]
Not sure yet; would need to wait for https://github.com/jkuhlmann/cgltf/issues/245 at a minimum.

**Reply by arpu** [reply]
any plans to look at this again? would be very useful and a officell https://github.com/KhronosGroup/glTF/blob/main/extensions/2.0/Khronos/KHR_animation_pointer/README.md

**Reply by arpu** [reply]
https://github.com/jkuhlmann/cgltf/pull/269

**Reply by zeux** [reply]
It'd need to wait for the PR you linked to get merged yeah. The extension is fairly general; parts of this extension might be difficult to implement. Some parts are probably reasonably easy to support, like materials, lights, cameras. Any parameter that's int/bool would require changes to core animation processing as well, but I suspect these are just a bad idea to animate for any reasonable renderer (e.g. doubleSided, UV index, etc.). UV offset/scale animation are harder to implement because of quantization. There might be some other things that are hard that I'm not thinking about OTOH.

**Reply by zeux** [reply]
Also it looks like three.js doesn't support this yet? There's a PR https://github.com/mrdoob/three.js/pull/24108 but it hasn't been merged. I might need to wait for that as well to be able to test anything here thoroughly... Not sure if I can use Babylon JS instead - but I'd assume you need three.js support at runtime as well to actually use that extension?

**Reply by arpu** [reply]
some progress on threejs https://github.com/mrdoob/three.js/pull/31761

**Reply by arpu** [reply]
https://github.com/needle-tools/three-animation-pointer
